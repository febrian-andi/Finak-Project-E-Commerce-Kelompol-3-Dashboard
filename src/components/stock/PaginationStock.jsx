import React from 'react';

const PaginationStock = ({ startIndex, endIndex, totalItems, rowsPerPage, currentPage, totalPages, setRowsPerPage, setCurrentPage }) => {
  return (
    <div className='flex items-center justify-between pt-4 mt-4 border-t border-gray-100'>
      <div className='text-sm text-gray-600'>{`${startIndex + 1}-${endIndex} of ${totalItems}`}</div>
      <div className='flex items-center gap-4'>
        <div className='flex items-center gap-2'>
          <span className='text-sm text-gray-600'>Rows per page:</span>
          <select
            value={rowsPerPage}
            onChange={(e) => {
              setRowsPerPage(Number(e.target.value));
              setCurrentPage(1);
            }}
            className='border rounded p-1'>
            <option value={20}>20</option>
            <option value={50}>50</option>
            <option value={100}>100</option>
          </select>
        </div>
        <div className='flex items-center gap-2'>
          <button onClick={() => setCurrentPage((prev) => Math.max(prev - 1, 1))} disabled={currentPage === 1} className='p-1 rounded disabled:opacity-50'>
            <svg className='w-5 h-5' viewBox='0 0 20 20' fill='currentColor'>
              <path fillRule='evenodd' d='M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z' clipRule='evenodd' />
            </svg>
          </button>
          <span className='text-sm text-gray-600'>
            {currentPage}/{totalPages}
          </span>
          <button onClick={() => setCurrentPage((prev) => Math.min(prev + 1, totalPages))} disabled={currentPage === totalPages} className='p-1 rounded disabled:opacity-50'>
            <svg className='w-5 h-5' viewBox='0 0 20 20' fill='currentColor'>
              <path fillRule='evenodd' d='M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z' clipRule='evenodd' />
            </svg>
          </button>
        </div>
      </div>
    </div>
  );
};

export default PaginationStock;
