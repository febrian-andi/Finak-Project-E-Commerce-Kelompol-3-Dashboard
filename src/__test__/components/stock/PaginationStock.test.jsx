import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import PaginationStock from '../../../components/stock/PaginationStock';

describe('PaginationStock Component', () => {
  const defaultProps = {
    startIndex: 1,
    endIndex: 20,
    totalItems: 100,
    rowsPerPage: 20,
    currentPage: 1,
    totalPages: 5,
    setRowsPerPage: jest.fn(),
    setCurrentPage: jest.fn(),
  };

  it('renders the correct item range and total items', () => {
    render(<PaginationStock {...defaultProps} />);
    expect(screen.getByText('1-20 of 100')).toBeInTheDocument();
  });

  it('renders the correct current page and total pages', () => {
    render(<PaginationStock {...defaultProps} />);
    expect(screen.getByText('1/5')).toBeInTheDocument();
  });

  it('calls setRowsPerPage and resets current page when rows per page is changed', () => {
    render(<PaginationStock {...defaultProps} />);

    const rowsPerPageSelector = screen.getByDisplayValue('20');
    fireEvent.change(rowsPerPageSelector, { target: { value: '50' } });

    expect(defaultProps.setRowsPerPage).toHaveBeenCalledWith(50);
    expect(defaultProps.setCurrentPage).toHaveBeenCalledWith(1);
  });

  it('matches snapshot', () => {
    const { container } = render(<PaginationStock {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
