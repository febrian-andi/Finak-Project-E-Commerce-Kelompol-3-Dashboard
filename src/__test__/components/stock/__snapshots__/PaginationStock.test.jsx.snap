// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationStock Component matches snapshot 1`] = `
<div>
  <div
    class="flex items-center justify-between pt-4 mt-4 border-t border-gray-100"
  >
    <div
      class="text-sm text-gray-600"
    >
      1-20 of 100
    </div>
    <div
      class="flex items-center gap-4"
    >
      <div
        class="flex items-center gap-2"
      >
        <span
          class="text-sm text-gray-600"
        >
          Rows per page:
        </span>
        <select
          class="border rounded p-1"
        >
          <option
            value="20"
          >
            20
          </option>
          <option
            value="50"
          >
            50
          </option>
          <option
            value="100"
          >
            100
          </option>
        </select>
      </div>
      <div
        class="flex items-center gap-2"
      >
        <button
          class="p-1 rounded disabled:opacity-50"
          disabled=""
        >
          <svg
            class="w-5 h-5"
            fill="currentColor"
            viewBox="0 0 20 20"
          >
            <path
              clip-rule="evenodd"
              d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <span
          class="text-sm text-gray-600"
        >
          1
          /
          5
        </span>
        <button
          class="p-1 rounded disabled:opacity-50"
        >
          <svg
            class="w-5 h-5"
            fill="currentColor"
            viewBox="0 0 20 20"
          >
            <path
              clip-rule="evenodd"
              d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
